app/controllers/scans_controller.rb

```diff
@@ -33,17 +33,30 @@ def current_request
 
   protected
 
+  def illiad_page
+    @illiad_page = { 'Action': '10', 'Form': '30', 'rft.genre': 'scananddeliverArticle' }
+  end
+
+  def illiad_item_data(scan)
+    @illiad_item_data = { 'rft.location': scan.origin,
+                          'rft.callnum': scan.holdings.first.callnumber,
+                          'rft.item': scan.holdings.first.barcode }
+  end
+
+  def illiad_citation(scan)
+    @illiad_params = { 'rft.jtitle': scan.item_title,
+                       'rft.au': scan.data[:authors],
+                       'rft.pages': scan.data[:page_range],
+                       'rft.atitle': scan.data[:section_title] }
+  end
+
+  def illiad_callback(scan)
+    @illiad_callback = { 'scan_referrer': successful_scan_url(scan) }
+  end
+
   def illiad_query(scan)
```

Comment:
Already started doing this on Chris’ suggestion. I noticed after trying rake that I also need to create a new spec file too.

> On Jun 12, 2015, at 9:52 AM, Jessie Keck <notifications@github.com> wrote:
> 
> In app/controllers/scans_controller.rb <https://github.com/sul-dlss/sul-requests/pull/179#discussion_r32333068>:
> 
> > +    @illiad_item_data = { 'rft.location': scan.origin,
> > +                          'rft.callnum': scan.holdings.first.callnumber,
> > +                          'rft.item': scan.holdings.first.barcode }
> > +  end
> > +
> > +  def illiad_citation(scan)
> > +    @illiad_params = { 'rft.jtitle': scan.item_title,
> > +                       'rft.au': scan.data[:authors],
> > +                       'rft.pages': scan.data[:page_range],
> > +                       'rft.atitle': scan.data[:section_title] }
> > +  end
> > +
> > +  def illiad_callback(scan)
> > +    @illiad_callback = { 'scan_referrer': successful_scan_url(scan) }
> > +  end
> > +
> >    def illiad_query(scan)
> This seems to be getting a bit larger than the context of the controller. It's possible that it may be time to refactor all these methods out of the controller class and into its own class or module.
> 
> —
> Reply to this email directly or view it on GitHub <https://github.com/sul-dlss/sul-requests/pull/179/files#r32333068>.
>